### PR TITLE
Build Blocks directly instead of using BlockBuilder in OrcRecordReader

### PR DESCRIPTION
**Pull request by Praveen2112**
In the above proposal (#12000) it was inferred that we can get some better performance when we try to populate the blocks directly (by populating the result on a native array and wrap a  corresponding `Block` on top of it) instead of going via `BlockBuilder` (provided the `DataType` and `Block` kind is known before hand). We try to implement the same model when we are creating `Block` from a orc file. These are the benchmark numbers that we obtained

```
NORMAL IMPL
Benchmark                                   Mode  Cnt  Score   Error  Units
BenchmarkStreamReaders.readBooleanNoNull    avgt   60  0.029 ± 0.004   s/op
BenchmarkStreamReaders.readBooleanWithNull  avgt   60  0.103 ± 0.014   s/op
BenchmarkStreamReaders.readByteNoNull       avgt   60  0.033 ± 0.003   s/op
BenchmarkStreamReaders.readByteWithNull     avgt   60  0.090 ± 0.009   s/op
BenchmarkStreamReaders.readDoubleNoNull     avgt   60  0.214 ± 0.002   s/op
BenchmarkStreamReaders.readDoubleWithNull   avgt   60  0.183 ± 0.002   s/op
BenchmarkStreamReaders.readFloatNoNull      avgt   60  0.175 ± 0.003   s/op
BenchmarkStreamReaders.readFloatWithNull    avgt   60  0.170 ± 0.001   s/op

OUR IMPL
Benchmark                                   Mode  Cnt  Score    Error  Units
BenchmarkStreamReaders.readBooleanNoNull    avgt   60  0.020 ±  0.001   s/op
BenchmarkStreamReaders.readBooleanWithNull  avgt   60  0.082 ±  0.002   s/op
BenchmarkStreamReaders.readByteNoNull       avgt   60  0.009 ±  0.001   s/op
BenchmarkStreamReaders.readByteWithNull     avgt   60  0.074 ±  0.003   s/op
BenchmarkStreamReaders.readDoubleNoNull     avgt   60  0.197 ±  0.002   s/op
BenchmarkStreamReaders.readDoubleWithNull   avgt   60  0.166 ±  0.003   s/op
BenchmarkStreamReaders.readFloatNoNull      avgt   60  0.160 ±  0.002   s/op
BenchmarkStreamReaders.readFloatWithNull    avgt   60  0.144 ±  0.001   s/op
```

### CODE REVIEW

**Comment by Praveen2112** [comment]
@kokosing , @tdcmeehan , @sopel39 Can you please review them ?

**Comment by findepi** [comment]
cc @dain

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/IntArrayBlockSupplier.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    public Block provideFromDirectStream(OrcDataSourceId dataSourceId, int batchSize, BooleanInputStream presentStream, LongInputStream dataStream)
+            throws IOException
+    {
+        if (presentStream == null) {
+            if (dataStream == null) {
+                throw new OrcCorruptionException(dataSourceId, "Value is not null but data stream is not present");
+            }
+            return populateNonNullBlock(batchSize, dataStream);
+        }
+        else {
```
Comment (electrum) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/IntArrayBlockSupplier.java]: We generally avoid having a "redundant else" (where the "if" block returns or throws unconditionally). IntelliJ has an inspection for this (under *Java | Control flow issues*).

Comment (Praveen2112) [reply presto-orc/src/main/java/com/facebook/presto/orc/reader/IntArrayBlockSupplier.java]: Have updated the code to the required format

**Pull request closed without merging**